packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -828,8 +828,7 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         VoterStake storage voterStake,
         address voterAddress
     ) internal {
-        if (slash + int256(voterStake.stake) > 0) voterStake.stake = uint256(int256(voterStake.stake) + slash);
-        else voterStake.stake = 0;
+        voterStake.stake = slash > 0 ? voterStake.stake + uint256(slash) : voterStake.stake - uint256(-slash);
```

Comment:
I'm removing the "penalty reduction" for the sake of the test but I think it's safer to keep it as I'm not 100% a WEI rounding could cause troubles.